src/main/java/io/cdap/e2e/pages/actions/CdfPluginPropertiesActions.java

```diff
@@ -987,4 +989,86 @@ public static void replaceValueInInputPropertyFromEnv(String pluginProperty, Str
       replaceValueInInputProperty(pluginProperty, envVariableKey);
     }
   }
+
+  /**
+   * Click on the Page Navigation button inside Plugin's properties page
+   *
+   * @param button ButtonName
+   */
+  public static void clickOnPageButton(String button) {
+    WaitHelper.waitForElementToBeClickable(CdfPluginPropertiesLocators.locateButton(button),
+                                           ConstantsUtil.DEFAULT_TIMEOUT_SECONDS);
+    ElementHelper.clickOnElement(CdfPluginPropertiesLocators.locateButton(button));
+  }
+
+  /**
+   * Select Source Plugin from the Replication list
+   *
+   * @param pluginName PluginName
+   */
+  public static void selectReplicationSourcePlugin(String pluginName) {
+    ElementHelper.clickOnElement(CdfPluginPropertiesLocators.locateSourcePluginNameInList(pluginName));
+  }
+
+  /**
+   * Click on Deploy the Replication pipeline
+   */
+  public static void deployReplicationPipeline() {
+      ElementHelper.clickOnElement(CdfPluginPropertiesLocators.deployReplicationPipeline);
+    }
+
+  /**
+   * Click on Start for the replication pipeline to start
+   */
+  public static void startReplicationPipeline() {
+    ElementHelper.clickIfDisplayed(CdfPluginPropertiesLocators.start, ConstantsUtil.DEFAULT_TIMEOUT_SECONDS);
+  }
+
+  /**
+   * Check for the pipeline status is running
+   */
+  public static void runTheReplicationPipeline() {
+    startReplicationPipeline();
+    WaitHelper.waitForElementToBeDisplayed(CdfPluginPropertiesLocators.running);
+  }
+
+  /**
+   * Open Advanced logs while pipeline is in running state
+   */
+  public static void openAdvancedLogs() {
+    CdfPluginPropertiesLocators.logs.click();
+    parentWindow = SeleniumDriver.getDriver().getWindowHandle();
+    ArrayList<String> tabs = new ArrayList(SeleniumDriver.getDriver().getWindowHandles());
+    SeleniumDriver.getDriver().switchTo().window(tabs.get(tabs.indexOf(parentWindow) + 1));
+    CdfPluginPropertiesLocators.advancedLogs.click();
+  }
+
+  /**
+   * Close the replication pipeline logs and stop the pipeline
+   */
+  public static void closeTheReplicationPipelineLogsAndStopThePipeline() {
+    SeleniumDriver.getDriver().switchTo().window(parentWindow);
+    ElementHelper.clickOnElement(CdfPluginPropertiesLocators.stop);
+  }
+
+  /**
+   * Verify Error message displayed on the header/at the top of the Plugin Properties page using the Error message
+   * location in the .properties file {@link ConstantsUtil#DEFAULT_ERROR_PROPERTIES_FILE}
+   *
+   * @param errorMessage Expected error message location
+   */
+  public static void verifyErrorMessage(String errorMessage) {
+      String expectedErrorMessage = PluginPropertyUtils.errorProp(errorMessage);
+      WaitHelper.waitForElementToBeDisplayed(CdfPluginPropertiesLocators.rowError);
```

Comment:
Is this message coming on header ?

Reply:
Yes, since the xpath is different we cannot use the framework step here.